community-website/src/community-project-boilerplate-docgen/src/widgets/configure.md

```diff
@@ -0,0 +1,27 @@
+---
+title: Widgets - Configure
+layout: widget.pug
+---
+
+# Configure
+
+## Description
+
+Configure is a widget that lets you provide raw search parameters to the Algolia API.
+
+Any of the searchParameters added to this widget will be forwarded to Algolia. For more information on the different parameters that can be set, have a look at the [reference](https://www.algolia.com/doc/api-client/javascript/search#search-parameters).
+
+## Options
+
+| Attributes           | Type       | Description
+| -                    | -          | -
+| `searchParameters`   | `Object`   | Search parameters to forward to Algolia
+
+## Code example
+
+```html
+<ais-configure
+  [searchParameters]="{ hitsPerPage: 3, analytics: true }"
```

Comment:
Is there some way in Angular to take all props? I'd prefer it if each prop would be a search parameter so you don't need to write an inline object

Reply:
Yes but you will need to define every props it could take in before so internally Angular transforms them as observables:

```js
@Input() hitsPerPage?: number;
@Input() analytics?: boolean;
@Input() ....
```

It would become a pain to maintain with the API.

Reply:
It's not about typings, the important piece of code I shown above is the `@Input()` and we need to write it in order to have AOT optimizations.

Reply:
Ahh I get it 😢